manifests/config.pp

```diff
@@ -69,7 +69,6 @@
 
   mcollective::config_file{"${mcollective::configdir}/server.cfg":
```

Comment:
server.cfg also not used anymore - I already removed the fact that used to read it.

Reply:
Hum. while looking the other places to adjust listed in your message bellow, I tried `--dm choria` and it seems that it's already affected by the server facts removal.  PuppetDB raise an exception with this query:

```
Performing PQL query: nodes[certname, deactivated] { (certname in inventory[certname] { facts.mcollective.server.collectives.match("\d+") = "mcollective" }) and (inventory {facts.os.family = "Debian"}) and (resources {type = "Class" and title = "Mcollective::Service"}) }
```

`inventory[certname] { facts.mcollective.server.collectives.match("\d+") = "mcollective" }` seems to be the culprit… and I can't find anything about this `.match("\d+")` which seems to be the root cause of the problem… :thinking:

Reply:
the `server` facts are gone, do you have the exception handy?

Reply:
The exception is triggered when using `.match()` on something that does not exist…

```
% puppet query 'inventory [certname] { facts.processors.inexistent.match("\d+") = "Intel(R) Xeon(R) E-2136 CPU @ 3.30GHz" }'
<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8"/>
<title>Error 500 javax.servlet.ServletException: org.postgresql.util.PSQLException: ERREUR: erreur de syntaxe sur ou pr�s de � ) �
  Position: 490</title>
</head>
<body><h2>HTTP ERROR 500 javax.servlet.ServletException: org.postgresql.util.PSQLException: ERREUR: erreur de syntaxe sur ou pr�s de � ) �
  Position: 490</h2>
<table>
<tr><th>URI:</th><td>/pdb/query/v4</td></tr>
<tr><th>STATUS:</th><td>500</td></tr>
<tr><th>MESSAGE:</th><td>javax.servlet.ServletException: org.postgresql.util.PSQLException: ERREUR: erreur de syntaxe sur ou pr�s de � ) �
  Position: 490</td></tr>
<tr><th>SERVLET:</th><td>-</td></tr>
<tr><th>CAUSED BY:</th><td>javax.servlet.ServletException: org.postgresql.util.PSQLException: ERREUR: erreur de syntaxe sur ou pr�s de � ) �
  Position: 490</td></tr>
<tr><th>CAUSED BY:</th><td>org.postgresql.util.PSQLException: ERREUR: erreur de syntaxe sur ou pr�s de � ) �
  Position: 490</td></tr>
</table>
<h3>Caused by:</h3><pre>javax.servlet.ServletException: org.postgresql.util.PSQLException: ERREUR: erreur de syntaxe sur ou pr�s de � ) �
  Position: 490
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:162)
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:767)
	at org.eclipse.jetty.server.handler.RequestLogHandler.handle(RequestLogHandler.java:54)
	at com.puppetlabs.trapperkeeper.services.webserver.jetty9.utils.MDCRequestLogHandler.handle(MDCRequestLogHandler.java:36)
	at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:173)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:500)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:547)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.postgresql.util.PSQLException: ERREUR: erreur de syntaxe sur ou pr�s de � ) �
  Position: 490
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2532)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2267)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:312)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:448)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:369)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:153)
	at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:103)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java)
	at puppetlabs.puppetdb.jdbc$call_with_array_converted_query_rows$fn__22264.invoke(jdbc.clj:319)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:814)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:741)
	at puppetlabs.puppetdb.jdbc$call_with_array_converted_query_rows.invokeStatic(jdbc.clj:311)
	at puppetlabs.puppetdb.jdbc$call_with_array_converted_query_rows.invoke(jdbc.clj:294)
	at puppetlabs.puppetdb.jdbc$call_with_array_converted_query_rows.invokeStatic(jdbc.clj:307)
	at puppetlabs.puppetdb.jdbc$call_with_array_converted_query_rows.invoke(jdbc.clj:294)
	at puppetlabs.puppetdb.query_eng$fn__31219$produce_streaming_body__31224$fn__31225$fn__31228$fn__31230$fn__31231.invoke(query_eng.clj:281)
	at puppetlabs.puppetdb.jdbc$with_transacted_connection_fn$fn__22384$fn__22385.invoke(jdbc.clj:512)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:771)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:741)
	at puppetlabs.puppetdb.jdbc$with_transacted_connection_fn$fn__22384.invoke(jdbc.clj:511)
	at puppetlabs.puppetdb.jdbc$retry_sql$attempt__22378.invoke(jdbc.clj:470)
	at puppetlabs.puppetdb.jdbc$retry_sql.invokeStatic(jdbc.clj:480)
	at puppetlabs.puppetdb.jdbc$retry_sql.invoke(jdbc.clj:461)
	at puppetlabs.puppetdb.jdbc$with_transacted_connection_fn.invokeStatic(jdbc.clj:510)
	at puppetlabs.puppetdb.jdbc$with_transacted_connection_fn.invoke(jdbc.clj:502)
	at puppetlabs.puppetdb.query_eng$fn__31219$produce_streaming_body__31224$fn__31225$fn__31228$fn__31230.invoke(query_eng.clj:280)
	at ring.util.io$piped_input_stream$fn__190.invoke(io.clj:28)
	at clojure.core$binding_conveyor_fn$fn__5754.invoke(core.clj:2030)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more
</pre>

</body>
</html>
FATAL: Last PuppetDB API response code 500
```

"syntax error at or near )"

Easily reproducible with:

```
% puppet query 'inventory [certname] { facts.foo.bar.match("\d+") = "baz" }'
```

Restoring `mcollective.server.collectives` should be fine!